src/main/java/solutions/fairdata/fdp/index/utils/IncomingPingUtils.java

```diff
@@ -0,0 +1,30 @@
+package solutions.fairdata.fdp.index.utils;
+
+import org.springframework.http.HttpEntity;
+import solutions.fairdata.fdp.index.entity.events.Event;
+import solutions.fairdata.fdp.index.entity.events.EventType;
+import solutions.fairdata.fdp.index.entity.events.IncomingPing;
+import solutions.fairdata.fdp.index.entity.http.Exchange;
+import solutions.fairdata.fdp.index.entity.http.ExchangeDirection;
+
+import javax.servlet.http.HttpServletRequest;
+
+public class IncomingPingUtils {
+
+    private static final EventType eventType = EventType.IncomingPing;
+
+    private static final Integer version = 1;
+
+    public static Event prepareEvent(HttpEntity<String> httpEntity, HttpServletRequest request) {
+        Event event = new Event();
+        event.setType(eventType);
+        event.setVersion(version);
+        event.setIncomingPing(new IncomingPing());
+        event.getIncomingPing().setExchange(new Exchange());
+        event.getIncomingPing().getExchange().setDirection(ExchangeDirection.INCOMING);
+        event.getIncomingPing().getExchange().setRemoteAddr(request.getRemoteAddr());
+        event.getIncomingPing().getExchange().getRequest().setFromHttpEntity(httpEntity);
+        event.getIncomingPing().getExchange().getRequest().setFromHttpServletRequest(request);
```

Comment:
The wall of text could be reduced with some intermediate variables.

```suggestion
        var ping = new IncomingPing();
        var ex = new Exchange();
        ex.setDirection(ExchangeDirection.INCOMING);
        ex.setRemoteAddr(request.getRemoteAddr());
        ex.getRequest().setFromHttpEntity(httpEntity);
        ex.getRequest().setFromHttpServletRequest(request);
        event.getIncomingPing().setExchange(ex);
        event.setIncomingPing(ping);
```

Could be reduced even further if some properties could be set through a constructor (I'm thinking mostly about the `direction` and `remoteAddr` properties for Exchange; but I don't have the overview where these changes could affect other code).